ejb/WFLY-6892-ejb-access-logging.adoc

```diff
@@ -0,0 +1,387 @@
+            </xs:annotation>
+            <xs:attribute name="name" type="xs:string" use="required"/>
+        </xs:complexType>
+
+        <xs:complexType name="valueType">
+            <xs:attribute name="value" use="required" type="xs:string"/>
+        </xs:complexType>
+
+        <xs:complexType name="pathType">
+            <xs:attribute name="relative-to" use="optional" type="xs:string"/>
+            <xs:attribute name="path" use="required" type="xs:string"/>
+        </xs:complexType>
+
+        <xs:complexType name="sizeType">
+            <xs:attribute name="value">
+                <xs:simpleType>
+                    <xs:restriction base="xs:string">
+                        <!-- XSD doesn't allow ^ or $ so ^[0-9]+[bkmgtp]?$ is invalid -->
+                        <xs:pattern value="[0-9]+[bkmgtp]"/>
+                    </xs:restriction>
+                </xs:simpleType>
+            </xs:attribute>
+        </xs:complexType>
+
+        <xs:complexType name="positiveIntType">
+            <xs:attribute name="value" use="required" type="xs:positiveInteger"/>
+        </xs:complexType>
+
+        <xs:complexType name="booleanValueType">
+            <xs:attribute name="value" use="required" type="xs:boolean"/>
+        </xs:complexType>
+
+        <xs:complexType name="propertiesType">
+            <xs:annotation>
+                <xs:documentation>
+                    A collection of free-form properties.
+                </xs:documentation>
+            </xs:annotation>
+            <xs:choice minOccurs="0" maxOccurs="unbounded">
+                <xs:element name="property">
+                    <xs:complexType>
+                        <xs:attribute name="name" type="xs:string" use="required"/>
+                        <xs:attribute name="value" type="xs:string" use="optional"/>
+                    </xs:complexType>
+                </xs:element>
+            </xs:choice>
+        </xs:complexType>
+
+        <xs:complexType name="formatterType">
+            <xs:annotation>
+                <xs:documentation>
+                    A formatter that can be assigned to an access log.
+                </xs:documentation>
+            </xs:annotation>
+            <xs:choice minOccurs="1" maxOccurs="1">
+                <xs:element name="simple-formatter" type="simpleFormatterType" maxOccurs="1"/>
+                <xs:element name="json-formatter" type="jsonFormatterType">
+                </xs:element>
+            </xs:choice>
+            <xs:attribute name="name" type="xs:string" use="required"/>
+        </xs:complexType>
+
+        <xs:complexType name="accessLogFormatterType">
+            <xs:annotation>
+                <xs:documentation>
+                    Defines a formatter.
+                </xs:documentation>
+            </xs:annotation>
+            <xs:choice minOccurs="1" maxOccurs="1">
+                <xs:element name="pattern-formatter" type="simpleFormatterType" maxOccurs="1"/>
```

Comment:
I would suggest just using the `named-formatter` and not include this one.

Reply:
Okay, I'll just use named-formatter for applying a formatter to a access-log type.  The inline formatter only takes pattern-formatter (not json-formatter), so removing it will make it more consistent without much loss.

Reply:
Thanks for the detailed review.  Now that a `formatter` can only be a `named-formatter`, would it be easier to just change it to a simpel attribute `formatter`?

Also a more general question is, I noticed jboss-as-logging.xsd tends to use value-type elemennt instead of simple attribute, is that the convention, or part of the limitation imposed by logmanager you mentioned above?  Would it be more concise if they are xml attributes in ejb access logging schema?

```xml
<console-access-log name="access-log1" formatter="formatter1" >
  <encoding value="UTF-8" />
</console-access-log>
```

vs

```xml
<console-access-log name="access-log1" >
  <encoding value="UTF-8" />
  <formatter>
    <named-formatter name="format1" />
  </formatter>
</console-access-log>
```

Reply:
Or we could remove the container `<formatter>` element to be something like:

```xml
<console-access-log name="access-log1" >
  <encoding value="UTF-8" />
  <named-formatter name="format1" />
</console-access-log>
```

Reply:
The main reason is backwards compatibility and the logging subsystem was the first subsystem written for JBoss AS 7 :) If I could I would completely re-write as we've learned a lot since then.

That said I would probably just go with `<formatter name="formatter/>`. One thing to consider too is if we want to use the logging terms. It likely makes sense, but we don't have to.

Reply:
I have no problem switching to logging terms like `file-handler` as opposed to `file-access-log`.